static/app/components/devtoolbar/components/replay/replayPanel.tsx

```diff
@@ -0,0 +1,126 @@
+import {smallCss} from '../../styles/typography';
+import PanelLayout from '../panelLayout';
+
+const TRUNC_ID_LENGTH = 16;
+const POLL_INTERVAL_MS = 3000;
+
+export default function ReplayPanel() {
+  const {projectSlug, projectId, projectPlatform, trackAnalytics, SentrySDK} =
+    useConfiguration();
+  const replay =
+    SentrySDK && 'getReplay' in SentrySDK ? SentrySDK.getReplay() : undefined;
+
+  const [replayId, setReplayId] = useState<string | undefined>(() =>
+    replay?.getReplayId()
+  );
+  // Polls periodically since a replay could be started by sessionSampleRate
+  useEffect(() => {
+    const intervalId = setInterval(
+      () => setReplayId(replay?.getReplayId()),
+      POLL_INTERVAL_MS
+    );
+    return () => clearInterval(intervalId);
+  }, [replay]);
+
+  const [isRecording, setIsRecording] = useState(
+    () => replay?.getReplayId() !== undefined
+  );
+  // Current replayId is only defined when there is an active recording session
+  useEffect(() => setIsRecording(replayId !== undefined), [replayId]);
+
+  // Used to persist the link to the last replay even if it's stopped. //TODO: this is lost after leaving the panel
+  const [lastReplayId, setLastReplayId] = useState<string | undefined>(() =>
+    replay?.getReplayId()
+  );
+  useEffect(() => {
+    if (replayId) {
+      setLastReplayId(replayId);
+    }
+  }, [replayId]);
+
+  return (
+    <PanelLayout title="Session Replay">
+      <Fragment>
+        <Button
+          size="sm"
+          icon={isRecording ? <IconPause /> : <IconPlay />}
+          disabled={!replay} // TODO: tooltip on why this is disabled,
+          onClick={() => {
+            if (replay) {
+              if (isRecording) {
+                replay.stop(); // stop should call flush
+              } else {
+                // If we're sampling errors, the we might be in buffering mode.
+                // In this case start() will do nothing. flush() will switch to session mode + start recording.
+                replay.flush();
+              }
+              setReplayId(replay.getReplayId()); // this will set isRecording
+            }
+          }}
+        >
+          {isRecording
+            ? t('In progress. Click to stop recording')
+            : t('Start recording the current session')}
+        </Button>
+        <div css={[smallCss, panelSectionCss, panelInsetContentCss]}>
+          {lastReplayId ? (
+            <span css={[resetFlexRowCss, {gap: 'var(--space50)'}]}>
+              {isRecording ? t('Current replay: ') : t('Last recorded replay: ')}
+              <SentryAppLink
+                to={{
+                  url: `/replays/${lastReplayId}`,
+                  query: {project: projectId},
+                }}
+                onClick={() => {
+                  trackAnalytics?.({
+                    eventKey: `devtoolbar.current-replay-link.click`,
+                    eventName: `devtoolbar: Current replay link clicked`,
+                  });
```

Comment:
can remove this because we already have `trackAnalytics` being called inside `<SentryAppLink>` already, we just need to set the context name by using `<AnalyticsProvider>`